src/frequenz/sdk/actor/power_distributing/power_distributing.py

```diff
@@ -379,6 +413,52 @@ async def _set_distributed_power(
 
         return self._parse_result(tasks, distribution.distribution, timeout_sec)
 
+    def _get_power_distribution(
+        self, request: Request, inv_bat_pairs: List[InvBatPair]
+    ) -> DistributionResult:
+        """Get power distribution result for the batteries in the request.
+
+        Args:
+            request: the power request to process.
+            inv_bat_pairs: the battery and adjacent inverter data pairs.
+
+        Returns:
+            the power distribution result.
+        """
+
+        def distribute_power_equally(
```

Comment:
`distribute_power_equally` is an inner function because it is called twice in the same method, it was added to avoid code duplication. It distributes "equally" or "blindly" the power between broken/unavailable batteries (batteries without metrics).

Initially the main motivation to keep it in the actor was because `DistributionAlgorithm` requires metrics to perform the distribution

Reply:
> Since we already have a separate DistibutionAlgorithm I think we could make it an abstract base class and have a SoCBalancingAlgorithm (the current one) and the new, say, BlindDivisionAlgorithm (using equally might be ambiguous because the current algorithm also says that it distributes power equally but the equality is the goal, that all ending SoC are equal).
I also wonder since this will be a breaking change, if we should add another level to power and have power.distribution.Algorithm, power.distribution.SocBalancingAlgorithm, etc. so you can use it like from ...power import distribution; distribution.SocBalancingAlgorithm.

These are both good points and IIUC this will be a breaking change if we want to have a proper public interface to distribute power. I think it's probably better to do this refactoring in a different PR.
Though I can try to "hack" the function `distribute_power()` to alternative accept a set of IDs instead of a list of components with metrics to perform a blind power distribution in `DistributionAlgorithm`. I'll add this change in a separate commit just in case is too hacky, and we want to remove it and wait for a proper refactoring instead.

Reply:
Making this refactor in a different PR is a fair point, but I would still move the `distribute_power_equally` function to `power._distribution_algorithm` in this PR, as it is in the end another distribution algorithm and makes the code in the actor more complex and difficult to follow.

I think is fine (and probably a good thing) to keep the more complex distribution algorithm separated from the dummy fallback algorithm, I wouldn't try to mix both. As I said, we can mix them at another level (another function that adds the extra logic to use one or the other based on some conditions).

Reply:
It would be more convenient if non-trivial changes are done in a separate high priority PR,  so that I can be unblocked for my battery pool changes.

Reply:
Sounds good to me!

Reply:
Created https://github.com/frequenz-floss/frequenz-sdk-python/issues/415 to address this conversation in a follow-up PR